SoftLayer/managers/dedicated_host.py

```diff
@@ -0,0 +1,363 @@
+
+        Example::
+
+        :param list tags: filter based on list of tags
+        :param integer cpus: filter based on number of CPUS
+        :param integer memory: filter based on amount of memory
+        :param string hostname: filter based on hostname
+        :param string disk: filter based on disk
+        :param string datacenter: filter based on datacenter
+        :param dict \\*\\*kwargs: response-level options (mask, limit, etc.)
+        :returns: Returns a list of dictionaries representing the matching
+                  dedicated host.
+
+
+
+        """
+        if 'mask' not in kwargs:
+            items = [
+                'id',
+                'name',
+                'cpuCount',
+                'diskCapacity',
+                'memoryCapacity',
+                'datacenter',
+                'guestCount',
+            ]
+            kwargs['mask'] = "mask[%s]" % ','.join(items)
+
+        call = 'getDedicatedHosts'
+
+        _filter = utils.NestedDict(kwargs.get('filter') or {})
+        if tags:
+            _filter['dedicatedHosts']['tagReferences']['tag']['name'] = {
+                'operation': 'in',
+                'options': [{'name': 'data', 'value': tags}],
+            }
+
+        if hostname:
+            _filter['dedicatedHosts']['name'] = (
+                utils.query_filter(hostname)
+            )
+
+        if cpus:
+            _filter['dedicatedHosts']['cpuCount'] = utils.query_filter(cpus)
+
+        if disk:
+            _filter['dedicatedHosts']['diskCapacity'] = (
+                utils.query_filter(disk))
+
+        if memory:
+            _filter['dedicatedHosts']['memoryCapacity'] = (
+                utils.query_filter(memory))
+
+        if datacenter:
+            _filter['dedicatedHosts']['datacenter']['name'] = (
+                utils.query_filter(datacenter))
+
+        kwargs['filter'] = _filter.to_dict()
+        func = getattr(self.account, call)
+        return func(**kwargs)
+
+    def get_host(self, host_id, **kwargs):
+        """Get details about a dedicated host.
+
+        :param integer : the host ID
+        :returns: A dictionary containing a large amount of information about
+                  the specified instance.
+
+        Example::
+
+            # Print out host ID 12345.
+            dh = mgr.get_host(12345)
+            print dh
+
+            # Print out only name and backendRouter for instance 12345
+            object_mask = "mask[name,backendRouter[id]]"
+            dh = mgr.get_host(12345, mask=mask)
+            print dh
+
+        """
+        if 'mask' not in kwargs:
+            kwargs['mask'] = (
+                'id,'
+                'name,'
+                'cpuCount,'
+                'memoryCapacity,'
+                'diskCapacity,'
+                'createDate,'
+                'modifyDate,'
+                'backendRouter[id, hostname, domain],'
+                'billingItem[id, nextInvoiceTotalRecurringAmount, '
```

Comment:
I'd think about reorganizing this to make it a little more readable, because it's not obvious the nesting structure of this mask (i.e. children and orderItem are under billingItem, but the brackets are hard to follow)

Maybe look at how masking is done where a docstring is used, where different indentation can be used to make nesting easier to read. The only problem with the docstring method is printing it out in the debugger can get a little difficult to read.